react/src/components/Chat/ChatMessages.tsx

```diff
@@ -0,0 +1,54 @@
+import Flex from '../Flex';
+import PureChatTokenCounter from './ChatTokenCounter';
+import VirtualChatMessageList from './VirtualChatMessageList';
+import { Message } from '@ai-sdk/react';
+import equal from 'fast-deep-equal';
```

Comment:
I got an error `Cannot find module 'fast-deep-equal' or its corresponding type declarations.`. 
And since WebUI is using lodash, it's better to use `_.isEqual` instead of installing `fast-deep-equal`.

If `fast-deep-equal` is needed, it should be added to package.json. This is because pnpm doesn't allow implicit dependencies.